packages/clerk-js/src/utils/querystring.ts

```diff
@@ -1,11 +1,54 @@
-import qs from 'qs';
-
 export const getQueryParams = (queryString: string) => {
-  return qs.parse(queryString || '', {
-    ignoreQueryPrefix: true,
-  }) as Record<string, string>;
+  const queryParamsObject: { [key: string]: string | string[] } = {};
+  const queryParams = new URLSearchParams(queryString);
+  queryParams.forEach((value, key) => {
+    if (key in queryParamsObject) {
+      // If the key already exists, we need to handle it as an array
+      const existingValue = queryParamsObject[key];
+      if (Array.isArray(existingValue)) {
+        existingValue.push(value);
+      } else {
+        queryParamsObject[key] = [existingValue, value];
+      }
+    } else {
+      queryParamsObject[key] = value;
+    }
+  });
+  return queryParamsObject as Record<string, string>;
+};
+
+type StringifyQueryParamsOptions = {
+  keyEncoder?: (key: string) => string;
 };
 
-export const stringifyQueryParams = (params: Record<string, unknown> | Array<unknown>) => {
-  return qs.stringify(params || {});
+export const stringifyQueryParams = (
+  params:
+    | Record<string, string | undefined | null | object | Array<string | undefined | null>>
+    | null
+    | undefined
+    | string,
+  opts: StringifyQueryParamsOptions = {},
+) => {
+  if (params === null || params === undefined) {
+    return '';
+  }
+
+  const queryParams = new URLSearchParams();
+  if (params && typeof params === 'object') {
+    Object.keys(params).forEach(key => {
+      const encodedKey = opts.keyEncoder ? opts.keyEncoder(key) : key;
+      const value = params[key];
+      if (Array.isArray(value)) {
+        value.forEach(v => v !== undefined && queryParams.append(encodedKey, v || ''));
+      } else if (value === undefined) {
+        return;
+      } else if (typeof value === 'object' && value !== null) {
+        queryParams.append(encodedKey, JSON.stringify(value));
+      } else {
+        queryParams.append(encodedKey, value || '');
```

Comment:
❓ In the case of `{key: null}`  a `key=` will be added in the querysting. Are we sure about this? Also, can we add a test for that?

Reply:
This was the behavior of `qs` and we decided to keep it the same. @EmmanouelaPothitou please add a test for this case as Dimitris pointed out